protocol/test-sequencing.md

```diff
@@ -0,0 +1,229 @@
+
+Another way might be to intercept the engine-API `engine_forkchoiceUpdated(fcState, attributes)`,
+which can then queue the block-building work and complete it when there is a matching test block-building job.
+This is similar to the sidecar design in OP-Stack.
+However, this then makes the op-node lead, and prevents reorgs from being initiated by the test tool.
+The "mock" L1 mode will thus be prioritized.
+
+#### Compared to op-wheel
+
+`op-wheel` may be time to deprecate. Full deprecation is out of scope for this design doc,
+but decision is made to not add additional things to `op-wheel`.
+
+The `op-wheel` CLI consists of `db` (patch geth DB contents) and `engine` utils.
+
+The forkchoice / reorg CLI tools are still used,
+but may be possible to move to op-node or op-geth.
+
+The `engine` CLI tools are mostly one-off commands,
+not suited to build blocks with particular transactions, nor a kurtosis standalone service.
+
+There is an `engine auto` command for automatic block-building of L1 blocks (no L2 system deposits)
+on an interval using the engine-API.
+And it provides none of the testing or production sequencer functionality.
+
+### Production
+
+What if we want to make the test sequencer a production sequencer?
+
+The following integration work is needed:
+- Use `auto` sequencing mode by default
+- Detect when op-node is deriving conflicting blocks, and automatically back-off.
+    - When the last seen parent-block is not the tip of the chain.
+    - When `unsafe == safe`, aka likely syncing from L1 data.
+- Host the same sequencer admin RPC as op-node, to hande sequencer rotation:
+    - `admin_startSequencer`: start
+    - `admin_stopSequencer`: stop
+    - `admin_sequencerActive`: sequencer state
+    - `admin_postUnsafePayload`: publish a block, for failover
+    - `admin_overrideLeader`: used by op-conductor
+    - `admin_conductorEnabled`: used by op-conductor
+- Connect to geth P2P, to read/write the shared tx-pool.
+- Support the DA-throttling RPC.
+  Or better, implement proper multi-block DA congestion safe-guards.
+
+### Engine mode
+
+To not only build blocks locally, we can make the service connect to a regular engine-API endpoint,
+and use that to build the blocks.
+
+To keep the op-node in sync, forkchoice updates will need to be directed at a new op-node RPC,
+to keep the op-node forkchoice in sync, while passing through the calls to the actual engine API.
+
+This won't support the test capabilities, but does enable migration away from op-node sequencing.
+
+The service essentially becomes an interface for the op-conductor setup to work with any EL builder setup.
+
+**This includes rollup-boost**: without modifications to rollup-boost, we can migrate sequencing away from the op-node.
+[Flashblocks utilizes a sidecar design](https://github.com/ethereum-optimism/design-docs/pull/86),
+called rollup-boost, that enables flash-blocks to be flexibly attached between the services.
```

Comment:
If you designed things from first principles, would it make sense to no longer run rollup boost as an intermediate proxy? I think its fine for now to keep things the way they are but concerned about tech debt coming from inability to coordinate an integration point

Reply:
+1 There are likely better ways to run rollup-boost on top of this if/when this reaches the production milestone, but I think for now that's out of scope. We can learn from prior milestones, and then adapt to rollup-boost/builder needs based on the learnings.